drivers/ieee802154/ieee802154_stm32wba.c

```diff
@@ -0,0 +1,1051 @@
+	struct net_pkt *pkt;
+	struct stm32wba_802154_rx_frame *rx_frame;
+	uint8_t pkt_len;
+
+	ARG_UNUSED(arg2);
+	ARG_UNUSED(arg3);
+
+	LOG_DBG("RX thread started");
+
+	while (1) {
+		pkt = NULL;
+		rx_frame = NULL;
+
+		rx_frame = k_fifo_get(&stm32wba_radio->rx_fifo, K_FOREVER);
+
+		__ASSERT_NO_MSG(rx_frame->psdu);
+
+		/* Depending on the net L2 layer, the FCS may be included in length or not */
+		if (IS_ENABLED(CONFIG_IEEE802154_L2_PKT_INCL_FCS)) {
+			pkt_len = rx_frame->length;
+		} else {
+			pkt_len = rx_frame->length -  IEEE802154_FCS_LENGTH;
+		}
+
+#if defined(CONFIG_NET_BUF_DATA_SIZE)
+		__ASSERT_NO_MSG(pkt_len <= CONFIG_NET_BUF_DATA_SIZE);
+#endif
+
+		LOG_DBG("Frame received - sequence nb: %u, length: %u", rx_frame->psdu[2],
+			pkt_len);
+
+		/* Block the RX thread until net_pkt is available, so that we
+		 * don't drop already ACKed frame in case of temporary net_pkt
+		 * scarcity. The STM32WBA 802154 radio driver will accumulate any
+		 * incoming frames until it runs out of internal buffers (and
+		 * thus stops acknowledging consecutive frames).
+		 */
+		pkt = net_pkt_rx_alloc_with_buffer(stm32wba_radio->iface, pkt_len,
+						   AF_UNSPEC, 0, K_FOREVER);
+
+		if (net_pkt_write(pkt, rx_frame->psdu, pkt_len)) {
+			LOG_ERR("Failed to write packet data");
+			goto drop;
+		}
+
+		net_pkt_set_ieee802154_lqi(pkt, rx_frame->lqi);
+		net_pkt_set_ieee802154_rssi_dbm(pkt, rx_frame->rssi);
+		net_pkt_set_ieee802154_ack_fpb(pkt, rx_frame->ack_fpb);
+
+#if defined(CONFIG_NET_L2_OPENTHREAD)
+		net_pkt_set_ieee802154_ack_seb(pkt, rx_frame->ack_seb);
+#endif
+
+		if (net_recv_data(stm32wba_radio->iface, pkt) < 0) {
+			LOG_ERR("Packet dropped by NET stack");
+			goto drop;
+		}
+
+		rx_frame->psdu = NULL;
+
+		if (LOG_LEVEL >= LOG_LEVEL_DBG) {
+			log_stack_usage(&stm32wba_radio->rx_thread);
+		}
+
+		continue;
+
+drop:
+		rx_frame->psdu = NULL;
+
+		net_pkt_unref(pkt);
+	}
+}
+
+static void stm32wba_receive_failed(stm32wba_802154_rx_error_t error)
+{
+	const struct device *dev = stm32wba_get_device();
+
+	enum ieee802154_rx_fail_reason reason;
+
+	if (error == STM32WBA_802154_RX_ERROR_NO_BUFFERS) {
+		reason = IEEE802154_RX_FAIL_NOT_RECEIVED;
+	} else {
+		reason = IEEE802154_RX_FAIL_OTHER;
+	}
+
+	if (IS_ENABLED(CONFIG_IEEE802154_STM32WBA_LOG_RX_FAILURES)) {
+		LOG_INF("Receive failed, error = %d", error);
+	}
+
+	stm32wba_data.last_frame_ack_fpb = false;
+	stm32wba_data.last_frame_ack_seb = false;
+
+	if (stm32wba_data.event_handler) {
+		stm32wba_data.event_handler(dev, IEEE802154_EVENT_RX_FAILED, (void *)&reason);
+	}
+}
+
+#if defined(CONFIG_NET_L2_CUSTOM_IEEE802154)
+static int stm32wba_configure_extended(enum ieee802154_stm32wba_config_type type,
+				       const struct ieee802154_stm32wba_config *config)
+{
+	switch (type) {
+	case IEEE802154_STM32WBA_CONFIG_CCA_THRESHOLD:
+		LOG_DBG("Setting CCA_THRESHOLD: %d", config->cca_thr);
+		if (st_ral_set_cca_energy_detect_threshold(config->cca_thr)) {
```

Comment:
If `st_ral_set_cca_energy_detect_threshold()` does not return an *essentially bolean* value, please compare against 0 (or any relevant value).

Ditto for the other calls to `st_ral_...()` with a return value.